ml_engine.py

```
@@ -1,149 +1,215 @@
+                       1 - (embedding <=> :query_vec) AS similarity
+                FROM song_metadata
+                WHERE embedding IS NOT NULL
+                ORDER BY embedding <=> :query_vec
+                LIMIT :k
+            """)
+            params = {"query_vec": vector_literal, "k": fetch_limit}
+
+        try:
+            result = db_session.execute(sql, params)
+            rows = result.fetchall()
+        except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)